src/entries/popup/hooks/swap/useSwapInputs.ts

```diff
@@ -248,6 +253,92 @@ export const useSwapInputs = ({
     [assetToBuyValue, independentField],
   );
 
+  const determineOutputCurrency = useCallback(
+    (asset: ParsedSearchAsset | null) => {
+      if (!asset) return null;
+
+      const { chainId } = asset;
+
+      const supportedChain = SUPPORTED_CHAINS.find(
+        (chain) => chain.id === chainId,
+      );
+
+      if (!supportedChain) return null;
+
+      const isNativeCurrencyEth =
+        supportedChain.nativeCurrency.symbol === 'ETH';
+
+      if (isNativeCurrencyEth && !isNativeAsset(asset.address, chainId)) {
+        // Return ETH for this chain
+        return {
+          address: 'eth',
+          chainId,
+          isNativeAsset: true,
+          ...supportedChain.nativeCurrency,
+        };
```

Comment:
we could be using `chainsNativeAsset[chainId]` to get the native asset from backend directly, isn't it?

Reply:
Oh yeah we could. Isn't `SUPPORTED_CHAINS` getting the chains from backend directly https://github.com/rainbow-me/browser-extension/blob/master/src/core/references/chains.ts#L22-L24 ?

Reply:
ah makes sense